### ISSUE TITLE
Running Problem

### ISSUE THREAD

**Issue by liumufenghua** [HY-Vision-Lab/FUNAD#3]
I am very interested in your work, but I have encountered some problems.

**Comment by liumufenghua** [comment]
I really appreciate the work you’ve done! But I have some problems while running demo test. I hope you can give me some advice.

Firstly, I find some demo parameters about paths that I can not understand them very well. 
    1-parser.add_argument('--data_path', type=str, default='/path/to/mvtec')
    2-parser.add_argument('--save_path', type=str, default='/path/to/save_path')
    3-parser.add_argument('--feature_path', type=str, default='/path/to/DINO')
    4-parser.add_argument('--synthetic_path', type=str, default='/path/to/sdas_perlin_10')
I can know the meaning of 1 and 2 parameter, but the other parameters are confusing me. How should I understand and modify them？
second, I find that there are many fixed paths in the code. Should I modify them while running the demo code？
example: class dataset(Dataset):
    def __init__(self, dataset_path='/home/jiin/Drive/dataset', data_name='MVTec_AD', score_path='/home/jiin/Drive/score',
                 class_name='bottle', is_train=True, pretrain=False, cropsize=224, resize=256, aug='mixup', threshold=0.5, noise_threshold=0.9):

Thank you for your time!

**Comment by hy-yhson** [comment]
Thank you for your interest in our work. 

Please download the dataset from the following links and specify the corresponding paths in your environment.

--feature_path: DINO feature (MVTec-AD) [[Our Link]](https://drive.google.com/file/d/12WaHebolz6Z2AqXXOIjVxVx94eij3MLF/view?usp=sharing)
--synthetic_path: Synthetic dataset (MVTec-AD) [[Our Link]](https://drive.google.com/file/d/19JvHr4rwuNaU6Dgmk-Kf9NwsuZvOmIMn/view?usp=sharing)

Also, the paths defined in the `dataset(Dataset)` will automatically update if it is correctly specified in `self_train_ad.py`.

**Comment by Shaurya026** [comment]
Hi @liumufenghua @hy-yhson, the `dataset(Dataset)` class itself doesn't seem to be used anywhere in the training code. We are using only the synthetic data's loading from the file `dataload.py`. 

@hy-yhson, want to understand were `cut-paste` and `mixup` operations used while training for more synthetic anomalies? apart from the perlin noise based anomalies given in **--synthetic_path: Synthetic dataset (MVTec-AD) [[Our Link]](https://drive.google.com/file/d/19JvHr4rwuNaU6Dgmk-Kf9NwsuZvOmIMn/view?usp=sharing)**

**Comment by hy-yhson** [comment]
@Shaurya026 Only the anomalies provided through our link were used for training.

**Comment by liumufenghua** [comment]
Dear Author,
When i run the train code, the code will get stuck, just like what I showed in the last picture. So, could you please give me some suggestions about this situation. Thanks very much.
![image](https://github.com/user-attachments/assets/eb13820c-85c2-4782-9ae1-79dab6dde0ed)

**Comment by liumufenghua** [comment]
Thanks for your help, I have been run through the training code. Now, I try my best to learn your whole code, and I find you seem to use the test dataset in training code as validation set. So, I think you use the best result as the final model and metrics. Am I right? Could you please help me clarify my doubts？
Thanks very much，and good luck to you！

**Comment by hy-yhson** [comment]
@liumufenghua Yes you are right, we save the model checkpoint achieving the highest evaluation metric during the training phase.

**Comment by liumufenghua** [comment]
> @liumufenghua Yes you are right, we save the model checkpoint achieving the highest evaluation metric during the training phase.

All right, thanks for your response in time.
